packages/dom/src/core/sensors/pointer/PointerSensor.ts

```diff
@@ -258,6 +288,33 @@ export class PointerSensor extends Sensor<
     }
   }
 
+  protected handleWindowScroll(event: Event, source: Draggable) {
+    if (!this.coordinates) return;
+
+    const frame = getFrameElement(source.element);
+
+    // TODO account for nested scrolls via frame utility
+    this.nestedScroll = {
+      x: frame?.ownerDocument.defaultView?.scrollX || 0,
+      y: frame?.ownerDocument.defaultView?.scrollY || 0,
+    };
+
+    if (this.manager.dragOperation.status.dragging) {
+      event.preventDefault();
+      event.stopPropagation();
+
+      console.log('scroll', this.coordinates, this.nestedScroll);
+
+      // TODO this clashes with autoscroller
+      this.manager.actions.move({
+        to: {
+          x: this.coordinates.x + this.nestedScroll.x,
+          y: this.coordinates.y + this.nestedScroll.y,
+        },
+      });
+    }
+  }
```

Comment:
Makes sense! I wanted to push up earlier rather than later since we already discussed keeping things out of PointerSensor via Slack.

These draft PRs are first attempts at identifying and resolving various issues I've encountered, and really I'm just intending to figure out the bounds a bit and get the conversation going. More than happy to keep iterating and finding the right solution.

FWIW, I also realised I don't actually need this for Puck. But I do need the auto scroller to react to iframe scrolling.